src/pretix/base/payment.py

```diff
@@ -539,38 +545,72 @@ def payment_form(self, request: HttpRequest) -> Form:
                 dates = [
-                    availability_date.datetime(se).date()
+                    availability_start.datetime(se).date()
                     for se in self.event.subevents.filter(
                         id__in=CartPosition.objects.filter(
                             cart_id=cart_id, event=self.event
                         ).values_list('subevent', flat=True)
                     )
                 ]
-                availability_date = min(dates) if dates else None
+                availability_start = min(dates) if dates else None
             elif self.event.has_subevents and order:
                 dates = [
-                    availability_date.datetime(se).date()
+                    availability_start.datetime(se).date()
                     for se in self.event.subevents.filter(
                         id__in=order.positions.values_list('subevent', flat=True)
                     )
                 ]
-                availability_date = min(dates) if dates else None
+                availability_start = min(dates) if dates else None
             elif self.event.has_subevents:
                 logger.error('Payment provider is not subevent-ready.')
                 return False
             else:
-                availability_date = availability_date.datetime(self.event).date()
+                availability_start = availability_start.datetime(self.event).date()
+            print(availability_start, now_dt)
 
-            if availability_date:
-                return availability_date >= now_dt.astimezone(tz).date()
+            if availability_start:
+                if availability_start > now_dt.astimezone(tz).date():
+                    return False
+
+        availability_end = self.settings.get('_availability_date', as_type=RelativeDateWrapper)
+        if availability_end:
+            if self.event.has_subevents and cart_id:
+                dates = [
+                    availability_end.datetime(se).date()
+                    for se in self.event.subevents.filter(
+                        id__in=CartPosition.objects.filter(
+                            cart_id=cart_id, event=self.event
+                        ).values_list('subevent', flat=True)
+                    )
+                ]
+                availability_end = min(dates) if dates else None
+            elif self.event.has_subevents and order:
+                dates = [
+                    availability_end.datetime(se).date()
+                    for se in self.event.subevents.filter(
+                        id__in=order.positions.values_list('subevent', flat=True)
+                    )
+                ]
+                availability_end = min(dates) if dates else None
+            elif self.event.has_subevents:
+                logger.error('Payment provider is not subevent-ready.')
+                return False
+            else:
+                availability_end = availability_end.datetime(self.event).date()
```

Comment:
dieser Codeblock ist identisch zu dem obendrüber, ich würde das gern in einen Helper auslagern statt duplizieren

Reply:
so hier zb:
https://github.com/pretix/pretix/compare/payment-available-from...payment-available-from2?expand=1
oO(was ist die eigentlich einfachste Möglichkeit größere Codevorschlage an einen PR zu posten?)

Reply:
I applied a patch pretty similar to yours, I made it a little more complex to highlight that using `min` in both cases is not an accident but a design choice.

I'm not aware of a better option to do this on GitHub :D